tgui/packages/tgui/interfaces/Plantmaster.js

```diff
@@ -0,0 +1,414 @@
+/**
+ * @file
+ * @copyright 2022
+ * @author Amylizzle (https://github.com/amylizzle)
+ * @license MIT
+ */
+
+import { useBackend, useSharedState, useLocalState } from "../backend";
+import { Button, Dimmer, Divider, Flex, NumberInput, Section, Box, Dropdown, Tabs, SectionEx, Stack, Table } from '../components';
+import { Window } from '../layouts';
+import { Fragment } from 'inferno';
+import { NoContainer, ReagentGraph, ReagentList } from './common/ReagentInfo';
+import { TableCell, TableRow } from "../components/Table";
+import { clamp } from 'common/math';
+import { capitalize } from './common/stringUtils';
+import { truncate } from '../format';
+
+export const Plantmaster = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    extractables,
+    seeds,
+    category,
+    category_lengths,
+    inserted,
+    inserted_container,
+    seedoutput,
+    splice_chance,
+    show_splicing,
+    splice_seeds,
+    sortBy,
+    sortAsc,
+  } = data;
+  return (
+    <Window
+      resizable
+      title="Plantmaster Mk4"
+      width={1100}
+      height={450}>
+      <Window.Content>
+        <Tabs>
+          <Tabs.Tab
+            selected={category === 'overview'}
+            onClick={() => {
+              act('change_tab', { 'tab': 'overview' });
+            }}>
+            Overview
+          </Tabs.Tab>
+          <Tabs.Tab
+            selected={category === 'extractables'}
+            onClick={() => {
+              act('change_tab', { 'tab': 'extractables' });
+            }}>
+            Seed Extraction {`(${category_lengths[0]})`}
+          </Tabs.Tab>
+          <Tabs.Tab
+            selected={category === 'seedlist'}
+            onClick={() => {
+              act('change_tab', { 'tab': 'seedlist' });
+            }}>
+            Seeds {`(${category_lengths[1]})`}
+          </Tabs.Tab>
+          <Tabs.Tab
+            backgroundColor={inserted_container !== null ? "green" : "blue"}
+            selected={inserted_container !== null}
+            icon="eject"
+            onClick={() => inserted_container !== null ? act('ejectbeaker') : act('insertbeaker')}
+            bold>
+            {inserted_container !== null ? inserted : "Insert Beaker"}
+          </Tabs.Tab>
+        </Tabs>
+
+
+        {category === 'overview' && <PlantOverview cat_lens={category_lengths} container={inserted ? inserted_container : null} seedoutput={seedoutput} />}
+        {category === 'extractables' && <PlantExtractables seedoutput={seedoutput} produce={extractables} sortBy={sortBy} sortAsc={sortAsc} />}
+        {category === 'seedlist' && <PlantSeeds seeds={seeds} seedoutput={seedoutput} splicing={show_splicing} splice_chance={splice_chance} splice_seeds={splice_seeds} sortBy={sortBy} sortAsc={sortAsc} />}
```

Comment:
Minor opinionated problem, but these go way beyond 80 characters on a single line and are just hard to parse quickly

This goes for everything in this file, I like to run the prettier formatter first and fix it up with the eslint auto-fix after to fix up any of tgui's eslint rules.

ESLint just kinda sucks at this stuff.

```suggestion
        {category === 'overview' && (
          <PlantOverview
            cat_lens={category_lengths}
            container={inserted ? inserted_container : null}
            seedoutput={seedoutput}
          />
        )}
        {category === 'extractables' && (
          <PlantExtractables seedoutput={seedoutput} produce={extractables} sortBy={sortBy} sortAsc={sortAsc} />
        )}
        {category === 'seedlist' && (
          <PlantSeeds
            seeds={seeds}
            seedoutput={seedoutput}
            splicing={show_splicing}
            splice_chance={splice_chance}
            splice_seeds={splice_seeds}
            sortBy={sortBy}
            sortAsc={sortAsc}
          />
        )}
```

Reply:
Honestly that looks so much less readable to me. I'mma leave it